eng/Generate.ps1

```diff
@@ -70,14 +71,42 @@ function Add-CadlRanch-TypeSpec([string]$testName, [string]$projectPrefix, [stri
         $configString = "--config=$configFile "
     }
     $projectDirectory = Join-Path $projectDirectory "src"
-    $tspMain = Join-Path $cadlRanchFilePath $testName "main.tsp"
-    $clientTsp = Join-Path $cadlRanchFilePath $testName "client.tsp"
-    $mainTypeSpecFile = If (Test-Path $clientTsp) { Resolve-Path $clientTsp } Else { Resolve-Path $tspMain }
+    $azureMainTsp = Join-Path $azureSpecsDirectory $testName "main.tsp"
+    $azureClientTsp = Join-Path $azureSpecsDirectory $testName "client.tsp"
+    $mainTsp = Join-Path $specsDirectory $testName "main.tsp"
+    $clientTsp = Join-Path $specsDirectory $testName "client.tsp"
+    
+    $entryFile = ""
+    if ($projectPrefix -eq "typespec-nonAzure-") {
+        $entryFile = $clientTsp
+        if (!(Test-Path $entryFile)) {
+            $entryFile = $mainTsp
+        }
+        if (!(Test-Path $entryFile)) {
+            $entryFile = $azureClientTsp
+        }
+        if (!(Test-Path $entryFile)) {
+            $entryFile = $azureMainTsp
+        }
```

Comment:
Why needs to fall back to azure tsp for non-azure projects?

Reply:
https://github.com/microsoft/typespec/issues/5400

Previous test specs are split into azure tsp and non-azure tsp. Therefore, some existing tests are generated from non-azure tsp. I don't want to delete these tests, so add the fall back logic here.